design/one-pager-merge-jets-proposal.md

```diff
@@ -0,0 +1,370 @@
+upgrading is not an option because of another bug or behavior change.
+
+With the mixed provider, we can change the underlying implementation to a
+hand-written controller whenever we face such a bug. With the Kubernetes API
+conversion tooling, we can manage the possible API changes as well. This would
+also mean that over time, we'd have less and less Terraform dependence in our
+ecosystem.
+
+### Confusion for Users and Maintainers
+
+Terraform is a great tool but in an ideal state, we'd like the cloud vendors to
+step in and maintain those providers. They may choose to keep using their TF
+provider but it's unlikely; AWS does not maintain their TF provider, Azure uses
+ARM in their TF provider and they've shown interest in generating Crossplane
+based on ARM and Google is using Magic Modules and DCL to generate TF provider,
+so it's likely that they will choose to do that for their Crossplane provider as
+well. [Provider Strategy doc][strategy-doc] goes into more detail about these
+tools. In summary, Terrajet will be a temporary solution that we put in place to
+get more coverage and at some point we'll need to discuss how to change the
+implementation to those tools.
+
+With that in mind, it's a hard choice to invest in separate Jet-based providers
+knowing that they will, even though after years, be deprecated at one point and
+users will have to migrate. At the same time, classic providers are very far
+from having the same coverage level. This creates confusion and reluctance to
+use either of the providers, which also results in decreased pace of
+contributions because of the divergence of efforts.
+
+With a single mixed provider, the story is much clearer; we fill the gaps with
+Terrajet and over time the implementation of the resources will change to be
+more _native_; be it cloud vendors using their own approaches or handwritten API
+calls community writes that may be necessary to overcome a bug in TF provider.
+As long as the CRD schema and the API behavior is same, the maintainers can make
+the best choice depending on the situation, communicate any breaking changes and
+provide a smooth experience to the users. In case cloud vendor steps in and
+decides to start its own provider, then the story is still clear; there is a new
+provider maintained by its vendor and you should migrate to that one once you
+see fit as opposed to Jet vs classic choice where you need to think about
+different trade-offs with different and uncertain timelines.
+
+### Ownership by The Cloud Vendors
+
+As stated above, we hope that cloud vendors will come in and maintain their
+Crossplane providers. It seems like Terraform has achieved this goal only for
+GCP; Azure and AWS Terraform providers are not maintained by the vendors. In our
+case, Crossplane is a CNCF project with an open governance model, so we can't
+really compare it with Terraform, a property of HashiCorp, in that respect but
+it still provides a data point.
```

Comment:
Are there other data-points apart from Terraform providers we could look to? Perhaps ACK, ASO, KCC, etc?